tests/test_verify_base_url.py

```diff
@@ -54,3 +60,21 @@ def test_disable_verify_via_env(testdir, httpserver, monkeypatch):
     httpserver.serve_content(content='<h1>Error!</h1>', code=500)
     result = testdir.runpytest('--base-url', httpserver.url)
     assert result.ret == 0
+
+
+def test_verify_retries_url(testdir, httpserver, monkeypatch):
```

Comment:
Looking at this again, I guess we would be repeating a test, specifically the test that a URL can't be resolved. I think removing it would be good but I really like your solution.